### ISSUE TITLE
/entityOperations/upsert breaks all stored data in version 4.0.2

### ISSUE THREAD

**Issue by vgonzalez7** [ScorpioBroker/ScorpioBroker#402]
Hi,

while updating entities via /entityOperations/upsert I found an issue that might corrupt the entire database. More specifically, it merges the new entity with all entities in the database. Here are the steps to reproduce:

**Step 1: Post some test entities**

```
curl --location 'http://<host>:<port>/ngsi-ld/v1/entityOperations/upsert' \
--header 'Content-Type: application/ld+json' \
--data-raw '[
    {
        "id": "testentity:temperature:1",
        "type": "Temperature",
        "value": {
            "type": "Property",
            "value": 22,
            "unitCode": "CEL"
        },
        "@context": "https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/temperature-context.jsonld"
    },
    {
        "id": "testentity:temperature:2",
        "type": "Temperature",
        "value": {
            "type": "Property",
            "value": 25,
            "unitCode": "CEL"
        },
        "@context": "https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/temperature-context.jsonld"
    },
    {
        "id": "testentity:fleet:1",
        "type": "FleetVehicleStatus",
        "speed": {
            "type": "Property",
            "value": 60,
            "unitCode": "KMH"
        },
        "@context": "https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/fleetvehiclestatus-context.jsonld"
    },
    {
        "id": "testentity:airq:2",
        "type": "AirQualityObserved",
        "no2": {
            "type": "Property",
            "value": 50,
            "unitCode": "GP"
        },
        "@context": "https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/airqualityobserved-context.jsonld"
    }
]'
```

**Step 1-2: Query all test entities**

At this point, you can retrieve any of these entities to check that everything worked properly.

**Step 2: Update some of the test entities**

```
curl --location 'http://<host>:<port>/ngsi-ld/v1/entityOperations/upsert' \
--header 'Content-Type: application/ld+json' \
--data-raw '[
    {
        "id": "testentity:fleet:1",
        "type": "FleetVehicleStatus",
        "speed": {
            "type": "Property",
            "value": 55,
            "unitCode": "KMH"
        },
        "@context": "https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/fleetvehiclestatus-context.jsonld"
    },
    {
        "id": "testentity:airq:2",
        "type": "AirQualityObserved",
        "no2": {
            "type": "Property",
            "value": 72,
            "unitCode": "GP"
        },
        "@context": "https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/airqualityobserved-context.jsonld"
    }
]'
```

**Step 2-2: Query all test entities**

For instance, let's retrieve one of the Temperature entities.
```
curl --location 'http://<host>:<port>/ngsi-ld/v1/entities/testentity%3Atemperature%3A1' \
--header 'Link: <https://raw.githubusercontent.com/SALTED-Project/contexts/main/wrapped_contexts/temperature-context.jsonld>;rel="http://www.w3.org/ns/json-ld#context";type="application/ld+json"' \
--header 'Accept: application/json+ld'
```

Here we can see that this entity, which we didn't even update, has been corrupted with values of the ones that we did update. In fact, even the ID and type have been replaced.
```
{
    "id": "testentity:airq:2",
    "type": "https://smartdatamodels.org/dataModel.Environment/AirQualityObserved",
    "https://smartdatamodels.org/dataModel.Environment/no2": {
        "type": "Property",
        "value": 72,
        "unitCode": "GP"
    },
    "https://smartdatamodels.org/dataModel.Transportation/speed": {
        "type": "Property",
        "value": 55,
        "unitCode": "KMH"
    },
    "value": {
        "type": "Property",
        "value": 22,
        "unitCode": "CEL"
    }
}
```

The database is corrupted too, having a look at the 'entity' table:
![imagen](https://github.com/ScorpioBroker/ScorpioBroker/assets/111740239/b5d1a09a-1962-4819-8159-2110db6b7e7d)
Which is very strange. Somehow the 'id' within the database remains correct but the 'id' of the NGSI-LD entity got replaced. You can see that all of the entities in the database got corrupted. This happens both with the 'replace' and the 'update' options of the '/entityOperations/upsert' endpoint.

Kind regards,
Víctor
@jlanza @lauramartingonzalezzz

PS: Also we had some problems with the entityOperations/query endpoint but we haven't tested it fully.

**Comment by ScorpioBroker** [comment]
Hi,

thanks for this. I got the issue fixed in #fix402. Merging soon.

**Comment by ScorpioBroker** [comment]
fixed in 4.0.3
